lib/addons/ebs-csi-driver/index.ts

```diff
@@ -1,49 +1,142 @@
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName
-        });
-    }
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+      addOnName: defaultProps.addOnName,
+      version: options?.version ?? defaultProps.version,
+      saName: defaultProps.saName,
+    });
+
+    this.ebsProps = {
+      ...defaultProps,
+      ...options,
+    };
+  }
+
+  providePolicyDocument(clusterInfo: ClusterInfo): PolicyDocument {
+    return getEbsDriverPolicyDocument(
+      clusterInfo.cluster.stack.partition,
+      this.options?.kmsKeys
+    );
+  }
+
+  async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const baseDeployment = await super.deploy(clusterInfo);
+
+    const cluster = clusterInfo.cluster;
+    let updateSc: KubernetesManifest;
+
+    if (this.ebsProps.storageClass) {
+      // patch resource on cluster
+    const patchSc =  new KubernetesPatch(cluster.stack, `${cluster}-RemoveGP2SC`, {
+        cluster: cluster,
+        resourceName: "storageclass/gp2",
+        applyPatch: {
+          metadata: {
+            annotations: {
+              "storageclass.kubernetes.io/is-default-class": "false",
+            },
+          },
+        },
+        restorePatch: {
+          metadata: {
+            annotations: {
+              "storageclass.kubernetes.io/is-default-class": "true",
+            },
+          },
+        },
+      });
 
-    providePolicyDocument(clusterInfo: ClusterInfo) : PolicyDocument {
-        return getEbsDriverPolicyDocument(clusterInfo.cluster.stack.partition, this.options?.kmsKeys);
+      // Create and set gp3 StorageClass as cluster-wide default
+      updateSc = new KubernetesManifest(
+        cluster.stack,
+        `${cluster}-SetDefaultSC`,
+        {
+          cluster: cluster,
+          manifest: [
+            {
+              apiVersion: "storage.k8s.io/v1",
+              kind: "StorageClass",
+              metadata: {
+                name: "gp3",
+                annotations: {
+                  "storageclass.kubernetes.io/is-default-class": "true",
+                },
+              },
+              provisioner: "ebs.csi.aws.com",
+              reclaimPolicy: "Delete",
+              volumeBindingMode: "WaitForFirstConsumer",
+              parameters: {
+                type: "gp3",
+                fsType: "ext4",
+                encrypted: "true",
+              },
+            },
+          ],
+        }
+      );
+
+      /* new scConstruct(cluster.stack, `${cluster}-PatchSC`, updateSc);
+      return Promise.resolve(updateSc);
+    } else {
+      const noOpConstruct = new Construct(cluster.stack, `${cluster}-NoOp`);
+      return Promise.resolve(noOpConstruct); */
+      patchSc.node.addDependency(baseDeployment);
+      updateSc.node.addDependency(baseDeployment);
+
+      return baseDeployment;
+    } else {
+        return baseDeployment;
     }

```

Comment:
Please format the code. There should not be two braces on the same level.

Reply:
Seems, my local formatter (Prettier is the cause for this and other formatting changes) - is there a recommended formatter to be used with this project, please let me know!